generator.py

```diff
@@ -5,56 +5,97 @@
 random.seed()
 
 
-def generate(note_threads_number, accord_threads_number, length, tonality, positivity):
+def generate(note_threads_number, accord_threads_number, length, tonality, positivity, tempo):
     """
     this function produces melody which contains set of melodies: common, and accord-melodies
     :param note_threads_number: (int)how many flows of common melodies do we want
     :param accord_threads_number: (int)how many flows of accord-melodies do we want
     :param length: (int)how many notes should be in the melody
     :param tonality: (list) it list of allowed sounds
     :param positivity: (int) 0 or 1 0-major, 1-minor
+    :param tempo: (int) speed of melody, more tempo -> more speed
     :return: (list of Melody) all the melodies which music contains
     """
-    seq_seq = list()
+    melody_sequence = list()
+    patternise_number = 3
+    main_melody = create_melody_core(tonality, positivity, length // 2 ** (patternise_number - tempo - 1) + 1, 2 ** (patternise_number - tempo))
 
+    for _ in range(0, patternise_number):
+        main_melody = double_notes(tonality, main_melody, positivity)
+
+    # cut melody
+    new_main_melody = melody.Melody(0)
+    for time in main_melody.note_sequence:
+        if time < length:
+            new_main_melody.note_sequence[time] = main_melody.note_sequence[time]
+    main_melody = new_main_melody
+
+    melody_sequence.append(main_melody)
+    accord_duration = 2
+    note_duration = 1
     for _ in range(0, accord_threads_number):
         i = 0
         this_accord_melody = melody.Melody(1)
-        while i < length:
-            a = random.randint(40, 70)
-            current_cluster = get_cluster(tonality, 1, 100, positivity, a)
+        while i * accord_duration < length:
+            a = main_melody.note_sequence[i].sounds[0]
+            current_cluster = get_cluster(tonality, accord_duration, 80, positivity, a)
```

Comment:
I'd add `volume=80` explicitly

Reply:
fixed
Now, volume, positivity and main note can be indicated explicitly